src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_connected_registry_delete(cmd,
+                                  client,
+                                  connected_registry_name,
+                                  registry_name,
+                                  yes=False,
+                                  resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    user_confirmation("Are you sure you want to delete the Connected Registry '{}' from '{}'?".format(
+        connected_registry_name, registry_name), yes)
+    try:
+        result = client.delete(resource_group_name, registry_name, connected_registry_name)
+        msg = "Connected registry successfully deleted. Please cleanup your sync tokens and scope maps. " + \
+              "Run the following commands for cleanup: \n\t" + \
+              "az acr token delete -n <sync token> -r {}\n\t".format(registry_name) + \
+              "az acr scope-map delete -n <sync scope map> -r {}\n".format(registry_name) + \
+              "Run the following command on all ascendency to remove the deleted registry gateway access: \n\t" + \
+              "az acr scope-map update -n <parent sync scope map> -r {} --remove-gateway {}".format(
+                  registry_name, " ".join([connected_registry_name] + DEFAULT_GATEWAY_SCOPE))
+        logger.warning(msg)
+        return result
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_connected_registry_deactivate(cmd,
+                                      client,
+                                      connected_registry_name,
+                                      registry_name,
+                                      yes=False,
+                                      resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    user_confirmation("Are you sure you want to deactivate the Connected Registry '{}' from '{}'?".format(
+        connected_registry_name, registry_name), yes)
+    return client.deactivate(subscription_id=subscription_id,
+                             resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             connected_registry_name=connected_registry_name)
+
+
+def acr_connected_registry_list(cmd,
+                                client,
+                                registry_name,
+                                parent=None,
+                                no_children=False,
+                                resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    connected_registry_list = list(client.list(resource_group_name, registry_name))
```

Comment:
will this go over all the pages if the list has continuation token?

Reply:
I'm not sure what do you mean. But this method goes through all the connected registries. That is the only way I can build the tree to display it.